src/lib/AnnotationControlsFSM.ts

```diff
@@ -41,20 +43,28 @@ export const stateModeMap = {
 export default class AnnotationControlsFSM {
     private currentState: AnnotationState;
 
+    private subscriptions: Subscription[] = [];
+
     constructor(initialState = AnnotationState.NONE) {
         this.currentState = initialState;
     }
 
     public getState = (): AnnotationState => this.currentState;
 
+    private notify = (): void => this.subscriptions.forEach(callback => callback(this.currentState));
+
+    public subscribe = (callback: Subscription): void => this.subscriptions.push(callback);
+
     public transition = (input: AnnotationInput, mode: AnnotationMode = AnnotationMode.NONE): AnnotationMode => {
         if (input === AnnotationInput.CLICK) {
             this.currentState = mode === stateModeMap[this.currentState] ? AnnotationState.NONE : modeStateMap[mode];
+            this.notify();
             return stateModeMap[this.currentState];
```

Comment:
Seems like this is a good example of why we typically avoid multiple returns. Should this be refactored?